crates/sqlexec/src/parser.rs

```diff
@@ -765,6 +920,22 @@ mod tests {
         }
     }
 
+    #[test]
+    fn create_credentials_roundtrips() {
+        let test_cases = [
+            "CREATE CREDENTIALS qa PROVIDER debug OPTIONS (table_type = 'never_ending')",
+            "CREATE CREDENTIALS qa PROVIDER debug OPTIONS (table_type = 'never_ending') COMMENT 'for debug'",
```

Comment:
After looking at this, I could potentially see PROVIDER causing some confusion since it someone could assume that PROVIDER is referencing some external service look hashicorp vault, and not the cloud provider we're trying to hook into.

I think this is good as-is, but wanted to bring it up in case there's any ideas for something better.

Reply:
Let's just keep the keyword `FROM`? Just like tunnels, `aws`, `gcs` etc. will just be `credential_type`.

Reply:
@scsmithr what do you think about this?

Reply:
Tbh I'd prefer `PROVIDER` over `FROM` here.